src/RelayArgumentsOfCorrectType.ts

```diff
@@ -0,0 +1,69 @@
+import { ValidationRule, FragmentSpreadNode, valueFromAST, ASTNode } from "graphql"
+import { GraphQLError, visit } from "./dependencies"
+import { getFragmentArgumentDefinitions } from "./argumentDefinitions"
+
+function findFragmentSpreadParent(nodes: readonly any[]): FragmentSpreadNode | undefined {
+  return nodes.find(isFragmentSpread)
+}
+
+function isFragmentSpread(node: any): node is FragmentSpreadNode {
+  return node != null && node.kind === "FragmentSpread"
+}
+
+function hasVariables(node: ASTNode): boolean {
+  let hasVars = false
+
+  visit(node, {
+    Variable() {
+      hasVars = true
+    },
+  })
+  return hasVars
+}
+
+export const RelayArgumentsOfCorrectType: ValidationRule = function RelayArgumentsOfCorrectType(context) {
+  return {
+    Directive(directive, _key, _parent, _path, ancestors) {
+      if (directive.name.value !== "arguments" || !directive.arguments) {
+        return false
+      }
+      const fragmentSpread = findFragmentSpreadParent(ancestors)
+      if (!fragmentSpread) {
+        return false
+      }
+      const fragmentDefinition = context.getFragment(fragmentSpread.name.value)
+      if (!fragmentDefinition) {
+        return false
+      }
+
+      const typedArguments = getFragmentArgumentDefinitions(context, fragmentDefinition)
+
+      directive.arguments.forEach(arg => {
+        const argDef = typedArguments[arg.name.value]
+        if (!argDef || !argDef.schemaType) {
+          return
+        }
+
+        const schemaType = argDef.schemaType
+
+        if (hasVariables(arg.value)) {
+          return
+        }
+
+        const value = valueFromAST(arg.value, schemaType)
+
+        if (value === undefined) {
+          context.reportError(
+            new GraphQLError(badValueMessage(fragmentDefinition.name.value, arg.name.value, schemaType.toString()), [
+              arg.value,
+            ])
```

Comment:
Just out of curiosity, is there any benefit of providing an array here instead of just the single node?

Reply:
Don't think there is one - this was just a copy/paste thing, will change.

Reply:
Ah that wasn’t necessary, just wondering for my own info. Thanks, tho!